social_core/backends/orcid.py

```diff
@@ -49,22 +57,48 @@ def get_user_details(self, response):
                     else:
                         email = emails_list[0].get('email', '')
 
-        return {'username': response.get('orcid'),
-                'email': email,
-                'fullname': fullname,
-                'first_name': first_name,
-                'last_name': last_name}
+        return {
+            'username': username,
```

Comment:
The `username` parameter might be unset here.

Reply:
This still applies - in case `orcid_identifier` is None (or empty), `username` is not set and the code will crash here.

Reply:
@nijel username is set to an empty string now...see line 33 of orcid.py:`first_name = last_name = email = username = ''` (correct me if I am wrong, but if username is empty like this - the backend will create a unique one "automagically"?)

Reply:
Ah sorry, I've missed that line.

Reply:
@nijel No problem. Tests are still ongoing...do we use pytest for this library?

Reply:
Yes, it uses pytest, see `tox.ini`.